tests/lib.rs

```diff
@@ -154,8 +142,15 @@ mod util {
             .add(Right::Lookup)
             .finalize().unwrap();
         rights.limit(&dir).unwrap();
-        capsicum::enter().unwrap();
-        let path = CString::new("lib.rs").unwrap();
-        let _ = dir.open_file(path, 0, None).unwrap();
+        let pid = unsafe { libc::fork() };
+        if pid == 0 {
+            capsicum::enter().unwrap();
+            let path = CString::new("lib.rs").unwrap();
+            let _ = dir.open_file(path, 0, None).unwrap();
+            unsafe { libc::_exit(0) };
```

Comment:
nit: Would you be up for adding a failure case? We could use [tempdir](https://docs.rs/tempfile/latest/tempfile/fn.tempdir.html) to create a directory which we wouldn't have access to, or do something like a lookup in `"./tests"`. We probably should avoid using `"./src"` and `"./tests"` which I think wouldn't work in some cases if the tests binary was run manually outside of the root crate directory.

Reply:
Yeah, I was noticing the same thing about "./src".

Reply:
It looks like there are other problems too.  We aren't monitoring the child process's status correctly. I suggest we merge this PR as-is, and then I'll do another that uses Nix as a dependency in order to use `waitpid` correctly, which will help fix these problems.

Reply:
:+1: works for me.